src/iasm.c

```diff
@@ -426,14 +426,10 @@ static OPND *asm_and_exp();
 static OPND *asm_cond_exp();
 static opflag_t asm_determine_operand_flags(OPND *popnd);
 static code *asm_genloc(Loc loc, code *c);
-static int asm_getnum();
+static long long asm_getnum();
```

Comment:
No implicit assumptions and strong couplings between different code locations please. At best don't even start to reason like that.
toInteger can return a type that can hold any integer of the target machine, so just use the return type of toInteger d_integer_t, see https://github.com/dlang/dmd/blob/2a1e3592d6d91c0d3eac6725584010ca3061422d/src/expression.h#L225.